index.js

```diff
@@ -1,14 +1,302 @@
+      Button({
+        type: 'link',
+        text: 'See live',
+        icon: 'link',
+        href: project.liveLink,
+      }),
+      Button({
+        type: 'link',
+        text: 'See Source',
+        icon: 'github-blue',
+        href: project.sourceCode,
+      }),
+    ],
+  });
+
+  const Divider = createComponent('div', {
+    className: 'divider',
+  });
+
+  const Right = createComponent('div', {
+    className: 'modal-right-block',
+    children: [Tags(project.tags), Divider, ModalFooter],
+  });
+
+  const Left = createComponent('div', {
+    className: 'modal-left-block',
+    children: [Paragraph(project.description)],
+  });
+
+  const ModalBodyContent = createComponent('div', {
+    className: 'modal-blocks mt-12',
+    children: [Left, Right],
+  });
+
+  const ModalBody = createComponent('div', {
+    className: 'modal-body',
+    children: [
+      Captions(project.captions),
+      Picture(project.image, 'modal-image'),
+      ModalBodyContent,
+    ],
+  });
+
+  const CloseButton = Button({
+    type: 'icon-button',
+    icon: 'cancel',
+  });
+
+  const ModalHeader = createComponent('div', {
+    className: 'modal-header mb-12',
+    children: [Title(project.title), CloseButton],
+  });
+
+  const ModalContent = createComponent('div', {
+    className: 'modal-content bg-n0',
+    children: [ModalHeader, ModalBody],
+  });
+
+  const Dialog = createComponent('div', {
+    className: 'modal-dialog',
+    children: [ModalContent],
+  });
+
+  const Modal = createComponent('div', {
+    className: 'modal',
+    children: [Dialog],
+  });
+
+  CloseButton.addEventListener('click', () => {
+    toggleModal();
+    modalContainer.removeChild(Modal);
+  });
+
+  return Modal;
+}
+
+/**
+ * Create and return a project card DOM Node,
+ * to be inserted into the project section on the main page.
+ */
+function createCard(project, invert) {
+  const CardImage = Picture(project.image);
+  const CardTitle = Title(project.title);
+  const CardText = Paragraph(project.description);
+  const ProjectCaptions = Captions(project.captions);
+  const ProjectTags = Tags(project.tags);
+  const CardButton = Button({ text: 'See Project' });
+
+  const CardFooter = createComponent('div', {
+    className: 'action',
+    children: [CardButton],
+  });
+
+  const CardBody = createComponent('div', {
+    className: `card-body mt-12 + ${invert ? ' swap' : ''}`,
+    children: [CardTitle, ProjectCaptions, CardText, ProjectTags, CardFooter],
+  });
+
+  const Card = createComponent('article', {
+    children: [CardImage, CardBody],
+    className: 'card',
+  });
+
+  CardButton.addEventListener('click', () => {
+    modalContainer.appendChild(createModal(project));
+    toggleModal();
+  });
+
+  return Card;
+}
+
+/**
+ * Add projects to HTML portfolio section
+ */
+function loadProjects(data = []) {
+  data.forEach((project, index) => {
+    portfolio.appendChild(createCard(project, index % 2 === 1));
+  });
+}
+
+loadProjects(DATA);
```

Comment:
We need to ensure that all the HTML of the section is created when the page loads. One way of doing this is:
```JavaScript
window.onload = () => {
  // The DOM elements creation or functions that create them are called here. i.e.
  loadProjects(DATA);
};

```